algorithms/src/Kokkos_Sort.hpp

```diff
@@ -200,7 +200,7 @@ class BinSort {
   bool sort_within_bins;
 
  public:
-  BinSort() {}
+  BinSort() = default;
```

Comment:
Remove.

Reply:
There is another constructor.

Reply:
So?  It's in `Imp::` and as far as I know we don't need it.  (Correct me if I am wrong)

Reply:
No, it's not in namespace `Impl`. 
The constructor was introduced in 62e4ec0179dd205b0f1553398bf35395b68f00d1 but I don't know the reason. I couldn't find any use of it on `GitHub` but would not want to break backward compatibility in this pull request.

Reply:
Fair enough.

@crtrott what was the rational for adding these?

Reply:
I don't remember ...

Reply:
Ok actually I think this was necessary to make some stuff working in LAMMPS I believe, where we actually may store a Sorter inside another class. And that class has a default constructor.